### PR TITLE
Update CnfCertificationSuiteRun.Status.Phase

### PR DESCRIPTION
**Pull request by shirmoran**
Update CnfCertificationSuiteRun.Status.Phase with the following phases:

- CreatingCertSuiteJob
- RunningCertSuite
- FailedToDeployCertSuitePod
- CertSuiteError
- CertSuiteFinished

### CODE REVIEW

File: controllers/cnfcertificationsuiterun_controller.go [hunk 1]
```diff
@@ -109,13 +115,19 @@ func (r *CnfCertificationSuiteRunReconciler) Reconcile(ctx context.Context, req
...
+
+	r.updateJobStatus(&cnfrun, "RunningCertSuite")
 	err := r.Create(ctx, cnfCertJobPod)
 	if err != nil {
 		log.Log.Error(err, "Failed to create CNF Cert job")
+		r.updateJobStatus(&cnfrun, "CertSuiteError")
+		return ctrl.Result{}, nil
 	}
 
+	r.updateJobStatus(&cnfrun, "CertSuiteFinished")
```
Comment (greyerof) [inline controllers/cnfcertificationsuiterun_controller.go:130]: I don't think this phase should be set here.

CertSuiteFinished should be set when the CNF program has finished (all the test cases have run correctly).

We need to separate the CNF job pod creation/deployment from the outcome of the CNF run itself.
- CreatingCertSuiteJob: The controller is deploying the CNF pod in the cluster. Line 118 is ok.
- New: FailedToDeployCertSuitePod: this should be set at line 126.

If the CNF Job pod was successfully deployed, now we need to update the phase depending on what happens when the CNF Cert Suite app is running the test cases.
- CertSuiteFinished: The Cert Suite finished running the test cases without any error/panic. This should be set by either:
  a) a go-routine in the controller that keeps watching the output of the cnf cert suite container (exit status). This go- 
routine could be fired at line 130, right before the `return ctrl.Result{}, nil`. 
  b) the side-car app. Not sure whether it's possible or not, but you could investigate on how to read the container or process exit output of one container from another container in the same pod. Some useful links: [share-process-namespace](https://kubernetes.io/docs/tasks/configure-pod-container/share-process-namespace/ ), [multi-container pods inter-communication](https://www.mirantis.com/blog/multi-container-pods-and-container-communication-in-kubernetes/)

Please, implement option (a) and investigate (b) if you want and have time, just for learning purposes.
  - CertSuiteError: The Cert Suite app finished with some error/panic. Not sure wheter we can determine this right now, as we might need some changes in the CNF Cert Suite code, but this phase should probably be set in the same go-routine as the CertSuiteFinished. Don't implement this yet, please. We can address setting this phase in further PRs.

**Review by greyerof — COMMENTED** [review]
Apart from the comment I've put, there's an easy improvement related to the "Run" CR phase that you can also implement in this PR: add the "status.Phase" as a print column for the output of the `oc get cnfcertificationsuiteruns` command. Please see the kubebuilder annotation `// +kubebuilder:printcolumn` [here](https://book.kubebuilder.io/reference/markers/crd.html). Some examples [here](https://www.brendanp.com/pretty-printing-with-kubebuilder/).

File: controllers/cnfcertificationsuiterun_controller.go [hunk 2]
```diff
@@ -109,13 +178,21 @@ func (r *CnfCertificationSuiteRunReconciler) Reconcile(ctx context.Context, req
 		cnfRunPodId, cnfrun.Spec.LabelsFilter, cnfrun.Spec.LogLevel, cnfrun.Spec.TimeOut)
 
 	// Launch the pod with the CNF Cert Suite container plus the sidecar container to fetch the results.
+	r.updateJobStatus(&cnfrun, "CreatingCertSuiteJob")
+	logrus.Info("Creating CNF Cert job pod")
 	config := cnfcertjob.NewConfig(cnfrun.Spec.LabelsFilter, cnfrun.Spec.LogLevel, cnfrun.Spec.ConfigMapName, cnfrun.Spec.PreflightSecretName)
 	cnfCertJobPod := cnfcertjob.New(config, podName)
+
+	r.updateJobStatus(&cnfrun, "RunningCertSuite")
```
Comment (greyerof) [inline controllers/cnfcertificationsuiterun_controller.go:186]: Maybe we should also move this RunningCertSuite status update to line 194, after the pod has been succesfully created?

**Review by greyerof — APPROVED** [review]
Good job!

File: controllers/cnfcertificationsuiterun_controller.go (hunk 2 above)
Comment (shirmoran) [reply controllers/cnfcertificationsuiterun_controller.go:186]: Yeah your'e right! I'll change it.

**Review by greyerof — APPROVED** [review]

**Pull request merged**